boards/shields/pmod_sd/doc/index.rst

```diff
@@ -0,0 +1,36 @@
+.. _pmod_sd:
+
+Digilent Pmod SD
+################
+
+Overview
+********
+
+The Pmod SD provides a full-sized SD card slot that can be accessed through SPI.
+
+Programming
+***********
+
+Set ``--shield pmod_sd`` when you invoke ``west build``. For example:
+
+.. zephyr-app-commands::
+   :zephyr-app: tests/drivers/disk/disk_access
+   :board: ek_ra8m1
+   :shield: pmod_sd
+   :goals: build
+
+References
+**********
+
+- `Pmod SD product page`_
```

Comment:
I think adding the PMOD pinout here might be useful as well:
![Screenshot 2024-12-19 at 12 08 46](https://github.com/user-attachments/assets/7dd88481-6ef4-4f8e-8ce0-962eb4fa5b2c)

Reply:
Sure, I added PMOD pinout and futures for the document.